### PR TITLE
=doc Consolidate path directives docs #1624

### PR DESCRIPTION
**Pull request by jlprat**
Refs #1624
Converts directoty to symlink
Consolidation of `path-directives.md`, that was only in the Java DSL, with `path-matchers.md`.

### CODE REVIEW

File: docs/src/main/paradox/scala/http/routing-dsl/path-matchers.md [hunk 1]
```diff
@@ -135,7 +135,78 @@ PathMatcher[L: Tuple](prefix: Path, extractions: L): PathMatcher[L]
 : Matches and consumes the given path prefix and extracts the given list of extractions.
 If the given prefix is empty the returned matcher matches always and consumes nothing.
 
+@@@
+
+@@@ div { .group-java }
+
+A path matcher is a description of a part of a path to match. The simplest path matcher is `PathMatcher.segment` which
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/routing-dsl/path-matchers.md]: This section was the one copied over from `path-directives.md`

**Comment by raboof** [comment]
I guess perhaps `path-matchers.md` could now be unified even further, but this is already great!

**Comment by jlprat** [comment]
Agree, `path-matchers.md` can be unified further by including the path combinators (equivalents of `|` and `~`).
I'll create a ticket for it.

**Comment by jlprat** [comment]
Ah, you've done it already: #1650

**Pull request merged**